File: Aug/Day22/find-the-minimum-area-to-cover-all-ones-i.java
//  Time: O(M * N)  ::  Space: O(1)

class Solution {
    public int minimumArea(int[][] grid) {
        int n = grid.length, m = grid[0].length;
        int minX = n, maxX = 0, minY = m, maxY = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (grid[i][j] == 1) {
                    minX = Math.min(minX, i);
                    maxX = Math.max(maxX, i);
                    minY = Math.min(minY, j);
                    maxY = Math.max(maxY, j);
                }
            }
        }

        return (maxX - minX + 1) * (maxY - minY + 1);
    }
}

File: Aug/Day22/find-the-minimum-area-to-cover-all-ones-i.py
#  Time: O(M * N)  ::  Space: O(1)
class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        n, m = len(grid), len(grid[0])
        minX, maxX = n, 0
        minY, maxY = m, 0

        for i in range(n):
            for j in range(m):
                if grid[i][j] == 1:
                    minX = min(minX, i)
                    maxX = max(maxX, i)
                    minY = min(minY, j)
                    maxY = max(maxY, j)

        return (maxX - minX + 1) * (maxY - minY + 1)